src/crepe/api/LoopManager.cpp

```diff
@@ -20,51 +22,47 @@ LoopManager::LoopManager() {
 	this->load_system<RenderSystem>();
 	this->load_system<ScriptSystem>();
 	this->load_system<InputSystem>();
+	this->event_manager.subscribe<ShutDownEvent>(
+		[this](const ShutDownEvent & event) { return this->on_shutdown(event); });
 }
 
-void LoopManager::process_input() { this->get_system<InputSystem>().update(); }
+void LoopManager::process_input() {
+	this->get_system<InputSystem>().update();
+	this->event_manager.dispatch_events();
+}
 
 void LoopManager::start() {
 	this->setup();
 	this->loop();
 }
-void LoopManager::set_running(bool running) { this->game_running = running; }
 
 void LoopManager::fixed_update() {
-	// TODO: retrieve EventManager from direct member after singleton refactor
-	EventManager & ev = this->mediator.event_manager;
-	ev.dispatch_events();
 	this->get_system<ScriptSystem>().update();
 	this->get_system<PhysicsSystem>().update();
 	this->get_system<CollisionSystem>().update();
 }
 
 void LoopManager::loop() {
-	LoopTimer & timer = this->loop_timer;
-	timer.start();
 
 	while (game_running) {
-		timer.update();
+		this->loop_timer.update();
 
-		while (timer.get_lag() >= timer.get_fixed_delta_time()) {
+		while (this->loop_timer.get_lag() >= this->loop_timer.get_fixed_delta_time()) {
 			this->process_input();
 			this->fixed_update();
-			timer.advance_fixed_update();
+			this->loop_timer.advance_fixed_update();
 		}
 
-		this->update();
+		this->frame_update();
```

Comment:
wordt de volgende scene iedere keer geladen aan het einde van de loop?

Reply:
ja dit was een vraag van max

Reply:
Waarom is dit? per frameupdate hoeven we toch niet de scene te laden?

Reply:
Er wordt niet ieder frame een nieuwe scène geladen, dat gebeurt alleen als er (in een Script) opdracht toe is gegeven. Het meerendeel van de frames zal deze functie dus niks doen, maar de functie moet wel ieder frame aan worden geroepen omdat ieder frame van scène gewisseld moet kunnen worden.

Reply:
Dankje max voor de uitleg